tern/extensions/scancode/executor.py

```diff
@@ -165,14 +165,17 @@ def add_file_data(layer_obj, collected_files):
     # we'll assume that we are merging the collected_files data with
     # the file level data already in the layer object
     logger.debug("Collecting file data...")
+    updated = False
     while collected_files:
         checkfile = collected_files.pop()
         for f in layer_obj.files:
             if f.merge(checkfile):
                 # file already exists and has now been updated
+                updated = True
                 break
-        # file didn't previously exist in layer so add it now
-        layer_obj.files.append(checkfile)
+        if not updated:
+            # file didn't previously exist in layer so add it now
+            layer_obj.files.append(checkfile)
```

Comment:
Can you check my logic here? I think this can be achieved this way:
```
for f in layer_obj.files <-- cycle through existing files
    merged = f.merge(checkfile) <-- try to update the file
    if not merged: <-- the file didn't get updated, means this is a unique file
        layer_obj.files.append(checkfile)
```

Reply:
I think that should work. I'll test it.

Reply:
I think we still need the `break` and to update the file only when the for loop has been broken or completed. Otherwise, with your proposal, say we have three files in the layer [A, B, C] and two files from scancode that needs updating: [A, C]. We go through this for loop and compare A to A, merged = True so file gets updated but then we move to the next f in layer files and compare B from the layer files to the same A from scancode... merged = False and so we add the file, even though it already exists and has been updated... and we are facing the same problem we have now. I think we could do:

```
merged = False <-- variable needs to exist outside of for loop so we can check it once the loop completes.
for f in layer_obj.files:
   merged = f.merge(checkfile)
   if merged:
      break
if not merged: <-- if we are here it's because we broke out of the loop (in which case the file has been updated and merged=True or because we never found a match and merged=False and file needs to be updated.
   layer_obj.files.append(checkfile)
```

Thoughts?

Reply:
Yup! This makes sense.